eng/build.sh

```diff
@@ -89,9 +89,19 @@ source $scriptroot/native/init-os-and-arch.sh
 # Check if an action is passed in
 declare -a actions=("b" "build" "r" "restore" "rebuild" "testnobuild" "sign" "publish" "clean")
 actInt=($(comm -12 <(printf '%s\n' "${actions[@]/#/-}" | sort) <(printf '%s\n' "${@/#--/-}" | sort)))
+i=0
 
 while [[ $# > 0 ]]; do
   opt="$(echo "${1/#--/-}" | awk '{print tolower($0)}')"
+
+  if [[ $i -eq 0 && $opt =~ ^[a-zA-Z\.\+]+ ]]; then
+    arguments="$arguments /p:Subset=$1"
+    shift 1
+    continue
+  fi
+
+  i=$(( i + 1 ))
+
```

Comment:
nit: could use a more descriptive naming of the variable as I don't think you need to count the arguments:

```suggestion
firstArgumentChecked=0

while [[ $# > 0 ]]; do
  opt="$(echo "${1/#--/-}" | awk '{print tolower($0)}')"

  if [[ $firstArgumentChecked -eq 0 && $opt =~ ^[a-zA-Z\.\+]+ ]]; then
    arguments="$arguments /p:Subset=$1"
    shift 1
    continue
  fi

  firstArgumentChecked=1
```

Reply:
done